src/cares_wrap.cc

```diff
@@ -1061,33 +1061,39 @@ int ParseSoaReply(Environment* env,
   EscapableHandleScope handle_scope(env->isolate());
   auto context = env->context();
 
-  /* Can't use ares_parse_soa_reply() here which can only parse single record */
+  // Manage memory using standardard smart pointer std::unique_tr
+  struct AresDeleter {
+    void operator()(char* ptr) const noexcept { ares_free_string(ptr); }
+  };
+  using ares_unique_ptr = std::unique_ptr<char[], AresDeleter>;
+
+  // Can't use ares_parse_soa_reply() here which can only parse single record
   unsigned int ancount = cares_get_16bit(buf + 6);
   unsigned char* ptr = buf + NS_HFIXEDSZ;
-  char* name;
-  char* rr_name;
+  char* name_temp;
   long temp_len;  // NOLINT(runtime/int)
-  int status = ares_expand_name(ptr, buf, len, &name, &temp_len);
+  int status = ares_expand_name(ptr, buf, len, &name_temp, &temp_len);
+  const ares_unique_ptr name(name_temp);
   if (status != ARES_SUCCESS) {
-    /* returns EBADRESP in case of invalid input */
+    // returns EBADRESP in case of invalid input
     return status == ARES_EBADNAME ? ARES_EBADRESP : status;
   }
 
   if (ptr + temp_len + NS_QFIXEDSZ > buf + len) {
-    free(name);
     return ARES_EBADRESP;
   }
   ptr += temp_len + NS_QFIXEDSZ;
 
+  char* rr_name_temp;
```

Comment:
could you move this inside the `for`? [ES.5](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-scope): Keep scopes small
also please declare a new `long temp_rr_len` and `int status2` [ES.26](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-recycle): Don’t use a variable for two unrelated purposes

Reusing variables has no runtime benefit, and has the risk of misuse.